cmd/packaging/packaging.go

```diff
@@ -162,15 +157,166 @@ func runDockerPackaging(path string, packagingFormat string, command []string) {
+			"version":             buildVersion,
+			"description":         pubspec.GetPubSpec().Description,
+			"organizationName":    androidmanifest.AndroidOrganizationName(),
+			"arch":                runtime.GOARCH,
+		}
+	})
+	return templateData
+}
+
+type packagingTask struct {
+	packagingFormatName            string
+	dependsOn                      map[*packagingTask]string
+	templateFiles                  map[string]string
+	executableFiles                []string
+	linuxDesktopFileExecutablePath string
+	linuxDesktopFileIconPath       string
+	dockerfileContent              []string
+	generateBuildFiles             func(projectName, path string)
+	buildOutputDirectory           string
+	packagingCommand               string
+	resultFileName                 string
+	outputFileExtension            string
+}
+
+func (t *packagingTask) Init(ignore ...bool) {
+	for f := range t.dependsOn {
+		switch f.packagingFormatName {
+		case DarwinBundlePackagingTask.packagingFormatName:
+			DarwinBundlePackagingTask.Init(true)
+		case DarwinDmgPackagingTask.packagingFormatName:
+			DarwinDmgPackagingTask.Init(true)
+		case DarwinPkgPackagingTask.packagingFormatName:
+			DarwinPkgPackagingTask.Init(true)
+		case LinuxAppImagePackagingTask.packagingFormatName:
+			LinuxAppImagePackagingTask.Init(true)
+		case LinuxDebPackagingTask.packagingFormatName:
+			LinuxDebPackagingTask.Init(true)
+		case LinuxRpmPackagingTask.packagingFormatName:
+			LinuxRpmPackagingTask.Init(true)
+		case LinuxSnapPackagingTask.packagingFormatName:
+			LinuxSnapPackagingTask.Init(true)
+		case WindowsMsiPackagingTask.packagingFormatName:
+			WindowsMsiPackagingTask.Init(true)
+		}
+	}
+	projectName := pubspec.GetPubSpec().Name
+	createPackagingFormatDirectory(t.packagingFormatName, ignore...)
+	dir := packagingFormatPath(t.packagingFormatName)
+	templateData := getTemplateData(projectName, "")
+	templateData["icon"] = executeStringTemplate(t.linuxDesktopFileIconPath, templateData)
+	templateData["exec"] = executeStringTemplate(t.linuxDesktopFileExecutablePath, templateData)
+	for sourceFile, destinationFile := range t.templateFiles {
+		destinationFile = executeStringTemplate(filepath.Join(dir, destinationFile), templateData)
+		err := os.MkdirAll(filepath.Dir(destinationFile), 0775)
+		if err != nil {
+			log.Errorf("Failed to create directory %s: %v", filepath.Dir(destinationFile), err)
+			os.Exit(1)
+		}
+		fileutils.ExecuteTemplateFromAssetsBox(fmt.Sprintf("packaging/%s", sourceFile), destinationFile, fileutils.AssetsBox, templateData)
+	}
+	createDockerfile(t.packagingFormatName, t.dockerfileContent)
+	log.Infof("go/packaging/%s has been created. You can modify the configuration files and add it to git.", t.packagingFormatName)
+	log.Infof(fmt.Sprintf("You now can package the %s using `%s`", strings.Split(t.packagingFormatName, "-")[0], log.Au().Magenta("hover build "+t.packagingFormatName)))
+}
+
+func (t *packagingTask) Build(buildVersion string) {
```

Comment:
`Build(..)` is a bit confusing because hover builds and packages.. How about `Pack(..)`?